### ISSUE TITLE
WFO1000 Security analyzer misreports if using own IComponent interface

### ISSUE THREAD

**Issue by stilldesign** [dotnet/winforms#12568]
### .NET version

dotnet 9.0

### Did it work in .NET Framework?

N/A

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

The analyzer is new to dotnet 9

### Issue description

dotnet 9 looks to of introduced a new WinForms security analyzer to help with serialization leaks: `WFO1000`
However this analyzer is triggering on classes which inherit from any interface named `IComponent`

This is an issue for me as I have an interface named `IComponent` as part of my implementation of the [Composite pattern](https://en.wikipedia.org/wiki/Composite_pattern)

### Steps to reproduce

```cs
namespace WinFormsApp1;

public class TestCase : WinFormsApp1.IComponent
{
  public string? Test { get; set; }
}

public interface IComponent
{
}
```

Will result in the analyzer triggering:

![Image](https://github.com/user-attachments/assets/17b53d99-f8db-4567-ae16-6abbc23b4af6)

**Comment by HeilOliver** [comment]
Each property is also marked in Blazor applications.

There is an IComponent interface in the Microsoft.AspNetCore.Components namespace which is implemented by the ComponentBase. This is inherited by almost all components in Blazor. 
The result is that every property is marked even when no winform is involved!

**Comment by KlausLoeffelmann** [comment]
Blazor is not supposed to pull in the WinForms Analyzers.
Do you have a Blazor-WinForms-Hybrid App where that's happening?

**Comment by davidwengier** [comment]
I just hit this today in my Winforms Blazor Hybrid application, upon upgrading to .NET 9.

**Comment by RussKie** [comment]
I think the problem is here:
https://github.com/dotnet/winforms/blob/93985d0759e96f6806c2432a91e39762fd9f6914/src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/CSharp/Analyzers/MissingPropertySerializationConfiguration/MissingPropertySerializationConfigurationAnalyzer.cs#L35

This check must only be conditioned for `System.ComponentModel.IComponent` interface. And (❕) it must be prefixed with `global::` to avoid false positives (for example see #7472).

**Comment by davidwengier** [comment]
Nah, you don't want to look at aliases in an analyzer, just a generator. The full type name is sufficient, and if anyone defines their own IComponent in the System namespace, they deserve their self inflicted foot wound :)

**Comment by KlausLoeffelmann** [comment]
This is a double of a part of #12476, therefore closing this.
